packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/widgets.dart';
+library;
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// The level of sensitivity that content in the widget tree has.
+///
+/// {@template flutter.services.ContentSensitivity}
+/// There are only three [ContentSensitivity] levels, and these can be set via a
+/// [SensitiveContent] widget.
+///
+/// [ContentSensitivity.sensitive] is the most severe setting, and if it is set,
+/// it will cause the tree to remain marked sensitive even if there are other
+/// `SensitiveContent` widgets in the tree.
+///
+/// [ContentSensitivity.autoSensitive] is the second most severe setting, and it will
+/// cause the tree to remain marked auto-sensitive if there are either (1) no other
+/// `SensitiveContent` widgets in the tree or (2) there are only other auto-sensitive
+/// or not sensitive `SensitiveContent` widgets in the tree.
+///
+/// [ContentSensitivity.notSensitive] is the least severe setting, and it will cause the
+/// tree to remain marked not sensitive as long as there are (1) no other `SensitiveContent`
+/// widgets in the tree or (2) there  are only other not sensitive `SensitiveContent` widgets
+/// in the tree. If there are no `SensitiveContent` widgets in the tree, the default setting as
+/// queried from the embedding will be used. This could be set by a Flutter developer in native
+/// Android; otherwise, Android uses [ContentSensitivity.autoSensitive] by default.
+/// {@endtemplate}
+///
+/// * See [SensitiveContent] for how to set a [ContentSensitivity] level
+///   in order for sensitive content to be obscured when the Flutter screen
+///   is shared.
+enum ContentSensitivity {
+  /// Content sensitivity is auto-detected by the native platform.
+  ///
+  /// When this level is set via a [SensitiveContent] widget, the window
+  /// hosting the screen will only be marked as sensitive if other [SensitiveContent]
+  /// widgets in the Flutter app with the [sensitive] level are present in the widget tree.
+  ///
+  /// See https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_AUTO for how
+  /// this mode behaves on native Android.
+  ///
+  /// This mode currently does not work for Flutter, and thus, will never obscure the screen during
```

Comment:
I think we should rephrase this. Maybe "This mode in android views attempts to auto detect passwords and 2factor tokens and other sensitive content. Android as of api 35 cannot determine if flutter views contain sensitive content automatically, and thus will never obscure the screen ..."